core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -918,18 +918,18 @@ public void addColumnMask(Table table, String column, Expression mask)
                 .map(entry -> {
                     NodeRef<Table> table = entry.getKey();
 
-                    List<ColumnInfo> columns = referencedFields.get(table).stream()
-                            .filter(field -> field.getName().isPresent()) // For DELETE queries, the synthetic column for row id doesn't have a name
-                            .map(field -> {
-                                String fieldName = field.getName().get();
-
-                                return new ColumnInfo(
-                                        fieldName,
-                                        columnMasks.getOrDefault(table, ImmutableMap.of())
-                                                .getOrDefault(fieldName, ImmutableList.of()).stream()
-                                                .map(Expression::toString)
-                                                .collect(toImmutableList()));
-                            })
+                    QualifiedObjectName tableName = entry.getValue().getName();
```

Comment:
Can the same table be added to `io.trino.sql.analyzer.Analysis#tableColumnReferences` twice with different column set? Is `io.trino.sql.analyzer.Analysis#tableColumnReferences` just for reporting purpose?

Reply:
I think, yes. They are mapped by `AccessControlInfo`, which might be different for a view and for the enclosing query.
`io.trino.sql.analyzer.Analysis#tableColumnReferences` is for access control checks for select. I used it because it identifies correctly all referenced table's columns.